[TSK-634] [TSK-679] [TSK-680] migration(ui-kit): course-container, become-a-coach-banner

ObitoDevv opened the pull request:
## Description

<!-- Provide a clear and concise description of the changes made in this pull request for UI-Kit components. -->

- Component name: `course-container` , `coach-banner`
- Figma link or screenshot: [Figma Design](https://www.figma.com/)

![image](https://github.com/user-attachments/assets/6147d18a-166e-4c3d-802c-69e5d582f2f8)

![image](https://github.com/user-attachments/assets/2f24da4c-463e-4475-9767-23f6ec2e59ce)



---

## Checklist

- [x] I have run the lint, build, and test commands locally, as explained in the README, before creating this PR to ensure the code is clean and functional.
- [x] My changes align with the **core models** that have been discussed and agreed upon.
- [x] The component design has been compared to the **Figma design** to ensure consistency.
- [x] The **component is fully responsive** and behaves as expected across different screen sizes.
- [x] The component **supports keyboard web accessibility** (if applicable).
- [x] I have written **unit tests** or **integration tests** to cover the component's functionality above the coverage threshold.
- [x] I have updated the **documentation** for the component, including usage examples and guidelines.
- [x] I have **commented** my code, particularly in complex or non-obvious areas.
- [x] The **component is properly exported** for use in other parts of the application.

---

## Related Issues

<!-- Link any related issue or task. For example: -->
Fixes #49

alebg reviewed (approved):
@ObitoDevv thanks! Looks good to me. Let's wait for @carlapbg 's review before merging.

One thing though, could you please link to the figma, or put a screenshot of the figma in the PR description, instead of a screenshot of the storybook? For this and for any future PR

carlapbg reviewed (dismissed):
Hi guys! Looks great to me. Thank you!

carlapbg commented on packages/ui-kit/lib/components/coach-banner.tsx:
```diff
@@ -0,0 +1,44 @@
+
+        <div className="order-1 md:order-2 h-full p-0 md:ml-10">
+          <img
+            src="https://res.cloudinary.com/dgk9gxgk4/image/upload/v1733464948/2151206389_1_c38sda.jpg"
```
Waiting for the client's feedback on whether this image should be hardcoded or no

alebg replied:
@ObitoDevv this image is still hard-coded, same as the title, subtitle, description. This is the main thing to refactor, since we need to be able to display different content in the coach banner

alebg reviewed (changes requested):
@ObitoDevv so in the end it turns out we need to be able to modify the `title`, `subtitle`, `description`, and image url of this component from the CMS, per platform. This implies that this component needs to be refactored: these three need to be passed as props to the componen from the outside, and not live in the translations package. So please proceed with these changes.

alebg commented on packages/ui-kit/stories/course-empty.stories.tsx:
```diff
@@ -0,0 +1,109 @@
+    ),
+  ],
+  argTypes: {
+    message: {
```
hi @ObitoDevv ! These story props (message and buttonText) are not really relevant for the empty state, right? If so, could you please remove them from the empty state story

ObitoDevv commented:
@alebg Sir, Thanks for the feedback! I've refactored the CoachBanner component as requested. The `title`, `subtitle`, `description`, and `imageUrl` are now passed as props from the outside, allowing them to be managed by the CMS per platform. They’ve been removed from the translations package. The button text and empty state remain in the translations package as discussed. I’ve also updated the Storybook configuration to reflect these changes, ensuring all props are properly typed and editable. Let me know if there’s anything else you’d like me to adjust!

alebg commented on packages/ui-kit/lib/components/coach-banner.tsx:
```diff
@@ -0,0 +1,60 @@
+      <div className="grid grid-cols-1 md:grid-cols-2 items-start w-full h-full sm:gap-3">
+        <div className="order-2 md:order-1 flex flex-col justify-start gap-4">
+          <h2 className="text-2xl font-bold leading-[150%] text-base-white">
+            {title}: <br className="hidden md:block" /> {subtitle}
```
@ObitoDevv please don't hardcode the ":" here, may end up being a nuisance

ObitoDevv replied:
@alebg Here are two option we can use : - 

Options - 

- We can remove this hardcoded `:` from the code and add in the title like this - 

![image](https://github.com/user-attachments/assets/7c3ac0c2-e16d-4123-a594-ab86ff705af5)

- We can add additional props for `sperator` and then we can use it. 

For me best option is to go with first one. Can you please tell me which should I prefer ?

alebg replied:
first one,as you already did, is good, thanks!

alebg commented on packages/ui-kit/lib/components/coach-banner.tsx:
```diff
@@ -0,0 +1,60 @@
+          <p className="text-gray-300 text-md md:text-base leading-[150%] mb-6">
+            {description}
+          </p>
+          <Button
```
@ObitoDevv how are we supposed to pass a callback to this button? Looking at this code, it looks like there's no way of making the button do anything. Please check

alebg reviewed (changes requested):
@ObitoDevv Please check my inline comments, about the hardcoded ":" between title and subtitle of coach banner, and the coach banner button that apparently cannot receive callbacks to manipulate what happens onClick

alebg reviewed (changes requested):
@ObitoDevv thanks a lot! this is almost ready to be merged, but one last thing for the sake of making sure everything works as it should, please add a test and a story that use the onClick callback with some example (can be a basic alert). 

Edit: @ObitoDevv on further reflection, the 'course-empty-state' should be combined with the 'course card list' component we made last week. So please migrate the course-empty-state code, with the latest improvements, to the course card list component, and make it such that, when you pass an empty list of courses to the course list component, then you show the empty state. Apologies if this wasn't clear from the figma or the tasks provided. 

For reference:
![immagine](https://github.com/user-attachments/assets/4bbe4341-17d3-4904-a4be-3b408c1ac4d1)

alebg commented:
- check image of coach banner, put placeholder if not working
- refactor course card list, port the empty state code here if the course list is empty

ObitoDevv commented:
@alebg Sir, I have pushed the code with updated course-card-component and new course-list-card component. Can you please verify it ?

alebg commented on packages/ui-kit/lib/components/coursecard/course-card.tsx:
```diff
@@ -9,75 +9,28 @@ import { TLocale } from '@maany_shr/e-class-translations';
   onDetails?: () => void;
   onBuy?: () => void;
-  // Coach/Visitor-specific props
+  onBrowseCourses?: () => void;
```
@ObitoDevv Some of these props are not being used in the CourseCard: onBrowseCourses, creatorName, groupName. If they are not being used, please cleanup this interface by removing them

ObitoDevv replied:
@alebg

Thanks for pointing out the unused props! I looked into `onBrowseCourses`, `creatorName`, and `groupName`. I’ve removed onBrowseCourses from CourseCardProps since it’s only used in CourseCardList for the EmptyState. I think creatorName was a typo for author, which is already handled via course.author. For groupName, I initially removed it, but since it’s passed from CourseCardList to CoachCourseCard, I’ve added it back to `CourseCardProps` as an optional prop to keep the props flow simple and avoid refactoring `TCourseMetadata`. The error Property `groupName` does not exist on type `CourseCardProps` is now resolved.

alebg replied:
@ObitoDevv my bad, groupName is being used. All good now, thanks

alebg commented on packages/ui-kit/lib/components/coursecard/course-list-card.tsx:
```diff
@@ -0,0 +1,89 @@
+  const baseCardProps = {
+    userType,
+    locale,
+    onEdit,
```
@ObitoDevv All these callbacks shouldn't be shared across cards. Each card should have its own callback, otherwise we'll have all cards' buttons doing exactly the same thing. Which doesn't make much sense

alebg commented on packages/ui-kit/lib/components/coursecard/course-list-card.tsx:
```diff
@@ -0,0 +1,89 @@
+  courses,
+  userType,
+  locale,
+  callbacks = {},
```
@ObitoDevv I.e., this prop (`callbacks`) is unnecessary

alebg commented on packages/ui-kit/stories/course-card/course-list-card.stories.tsx:
@ObitoDevv same here, please rename all files "course-list-card*" to "course-card-list*"

alebg commented on packages/ui-kit/stories/course-card/course-list-card.stories.tsx:
@ObitoDevv Something strange happens in this story. I can't modify the `courses` prop such that if I delete everything there, I see the empty state; nor I can modify the courses I see in the story

alebg commented on packages/ui-kit/lib/components/coach-banner.tsx:
```diff
@@ -0,0 +1,87 @@
+        </div>
+
+        <div className="order-1 md:order-2 h-full p-0 md:ml-10">
+          {isImageError ? (
```
@ObitoDevv thanks, this is working good now. One detail, can you also do the same thing in the case that the imageurl is empty? Otherwise we see this

![image](https://github.com/user-attachments/assets/7bf19c27-ccab-4044-b9ed-81dca5c9128d)

alebg replied:
@ObitoDevv we still see this, please address this as already pointed out last month

alebg replied:
@ObitoDevv This is still hapenning. Could you please coordinate with the rest of your team? This was already solved in the other components we merged

alebg reviewed (changes requested):
@ObitoDevv please check my in-line comments

carlapbg reviewed (changes requested):
Hi! I think I just have one observation: In the Course Card, can you please add a hover in the course name in case is too long?

carlapbg reviewed (changes requested):
Hi! In general it looks good. Just in the course card remember to add the hover on the title in case this is too long

alebg commented on libs/ui-kit/tests/course-card.test.tsx:
@ObitoDevv and this file...

alebg commented on packages/ui-kit/lib/components/coursecard/coach-coursecard/coach-coursecard.tsx:
@ObitoDevv one thing i've just noticed, is that the CourseCreator component doesn't receive a callback. So in the story it looks clickable, but in reality it isn't. Could you add this callback to be passed from the outside please? Otherwise we cannot control what clicking on the course creator name does

![immagine](https://github.com/user-attachments/assets/af2cf98e-4270-4750-af57-95400872227b)

alebg commented on packages/ui-kit/lib/components/coursecard/course-empty-state.tsx:
```diff
@@ -0,0 +1,79 @@
+        <div className="flex justify-center items-center w-full p-10 rounded-2xl border border-solid bg-card-fill border-card-stroke max-md:px-5">
+            <div className="flex flex-col justify-center items-center self-stretch my-auto text-center">
+                {message && <p className="text-base text-white">{message}</p>}
+                {buttonText?.trim() && onButtonClick && (
```
@ObitoDevv could you please pass a callback in the stories, so we can see this component in its full form? Right now I see it without a button 
![immagine](https://github.com/user-attachments/assets/f95fc184-b915-435c-a2fe-74d961ca2cb7)

alebg commented on packages/ui-kit/lib/components/coursecard/course-card.tsx:
@ObitoDevv all course cards show the alt text when there's no image, while, as discussed, they shouldn't show anything:

![immagine](https://github.com/user-attachments/assets/0a033fb0-93b9-4811-bda0-f438b87f9e43)

ObitoDevv replied:
@alebg  Sir , I have already implemented this thing and it's working fine in my system. Can you please do one thing to run this command `pnpm nx run translations:build`  then check this issue? I guess it will work.  


![image](https://github.com/user-attachments/assets/050392e3-d500-4141-ae2e-c5661d5260f6)

alebg replied:
@ObitoDevv I have everything up to date and build. What you need to do to trigger what I get is the following:

1. remove the imageUrl from the card and press Enter:
![immagine](https://github.com/user-attachments/assets/cd140fdc-fe25-4144-9539-cbc14b0a3938)

2. click on update story when prompted: 
![immagine](https://github.com/user-attachments/assets/07d97a19-9d3f-426c-818b-c5184586af0c)

If not prompted, then you can just click on "remount component:
![immagine](https://github.com/user-attachments/assets/562dd4b9-f395-4c78-adff-03b86abde9d3)

3. The component will reload and you'll see the alt text 
![immagine](https://github.com/user-attachments/assets/6b7a6cc2-90ef-4cd0-b012-a8b5dde0189a)

If you don't get it, please then try to follow these steps in Firefox.

ObitoDevv commented on libs/ui-kit/lib/components/coursecard/course-card.tsx:
@alebg Extremly sorry sir, for these files I don't know how these files came into the root dir because I haven't created these files. I will remove these files on next commit.

ObitoDevv commented:
@alebg Sir , 
Thanks a lot for the review. I have addressed all the above issues as per the comments. I would request you to please review it now. And Let me know if there’s anything else you’d like me to adjust!

alebg reviewed (changes requested):
@ObitoDevv apart from the image issues i've pointed out, the story for the student empty state is passing the emptystate component as a child. Which doesnt make much sense:
![immagine](https://github.com/user-attachments/assets/e12b4c7f-c488-4561-a065-a694ead58ad3)

As this is handled automatically by the CourseCardList component:
![immagine](https://github.com/user-attachments/assets/1ffc8a3b-adf3-4b62-b851-6b75fa6ffe75)

However, if I remove the unnecessary code from your story:
![immagine](https://github.com/user-attachments/assets/43b6a746-802b-41db-815c-70943699f37f)

Then even if I pass the empty state callback, I don't see any button in the story:
![immagine](https://github.com/user-attachments/assets/0f8e765e-65bf-4575-ae33-c61caf5841be)


It seems to me you have coded unnecessary complexity:
![immagine](https://github.com/user-attachments/assets/a10aec26-9010-4f5e-8372-6acbea1cdff3)


Just make the callback for the empty state mandatory, and *always* render the button. I.e., we dont need to conditionally render it like this:
![immagine](https://github.com/user-attachments/assets/f0c675c1-dcfc-438a-a8f7-e79c92ecf957)
Just show it always

anilrise11 commented:
> @ObitoDevv apart from the image issues i've pointed out, the story for the student empty state is passing the emptystate component as a child. Which doesnt make much sense: ![immagine](https://private-user-images.githubusercontent.com/62880920/435406600-e12b4c7f-c488-4561-a065-a694ead58ad3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxMzEwNDMsIm5iZiI6MTc0NTEzMDc0MywicGF0aCI6Ii82Mjg4MDkyMC80MzU0MDY2MDAtZTEyYjRjN2YtYzQ4OC00NTYxLWEwNjUtYTY5NGVhZDU4YWQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDA2MzIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmY2E4NmIwNGM0MzdhZGVmMGM0ZjhhM2Y5ODU5M2IwYjYxYmNmMWY1Y2IwYmViMTdhNDZhMGFjMDEwMWRiNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2ARvWNjtjbsl89NyKeC6Yhb9iunBh4oled6LykY7dJ8)
> 
> As this is handled automatically by the CourseCardList component: ![immagine](https://private-user-images.githubusercontent.com/62880920/435406640-1ffc8a3b-adf3-4b62-b851-6b75fa6ffe75.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxMzEwNDMsIm5iZiI6MTc0NTEzMDc0MywicGF0aCI6Ii82Mjg4MDkyMC80MzU0MDY2NDAtMWZmYzhhM2ItYWRmMy00YjYyLWI4NTEtNmI3NWZhNmZmZTc1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDA2MzIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRhNDFhZjM3ZmI4Njc5MDQzZTg5MjUwMzdiNzcxMzhkZmY0MmNiYjA0MWUwZWExZDRkN2FlODVhNDJkODA3MWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.zhxJ_2adPWnv6_QGo6tpvwa2xEEgKM47bZJABlv7_Rk)
> 
> However, if I remove the unnecessary code from your story: ![immagine](https://private-user-images.githubusercontent.com/62880920/435406657-43b6a746-802b-41db-815c-70943699f37f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxMzEwNDMsIm5iZiI6MTc0NTEzMDc0MywicGF0aCI6Ii82Mjg4MDkyMC80MzU0MDY2NTctNDNiNmE3NDYtODAyYi00MWRiLTgxNWMtNzA5NDM2OTlmMzdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDA2MzIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmNzc4ZTlhYTNkNTczZGFkM2EyOWRhNTQzNTFjMmJiNzY1NzhlZjkxZGFkZDZmZWY3M2Y2Nzc3NDAxYTRiNGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.n6flKo51G9Ot5aHSdpHClI44QCeFDT0USFrtnwCXZZY)
> 
> Then even if I pass the empty state callback, I don't see any button in the story: ![immagine](https://private-user-images.githubusercontent.com/62880920/435406752-0f8e765e-65bf-4575-ae33-c61caf5841be.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxMzEwNDMsIm5iZiI6MTc0NTEzMDc0MywicGF0aCI6Ii82Mjg4MDkyMC80MzU0MDY3NTItMGY4ZTc2NWUtNjViZi00NTc1LWFlMzMtYzYxY2FmNTg0MWJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDA2MzIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhNmE2NzhkY2JjNTIxYTNiMTZhYTA5Mzc5ZTllMjY1NGE4YjA4NGQwMWZiMjNmM2Q3ZDU5MWI0NGQ3MTNhYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.97gaieWDCZ6aiAAVAtJosm2ssk6QDMgtdGJmdg3zIXM)
> 
> It seems to me you have coded unnecessary complexity: ![immagine](https://private-user-images.githubusercontent.com/62880920/435406794-a10aec26-9010-4f5e-8372-6acbea1cdff3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxMzEwNDMsIm5iZiI6MTc0NTEzMDc0MywicGF0aCI6Ii82Mjg4MDkyMC80MzU0MDY3OTQtYTEwYWVjMjYtOTAxMC00ZjVlLTgzNzItNmFjYmVhMWNkZmYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDA2MzIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMyZDQ1MWQxNjM0YzI5MDY4NTUzNTg3OGM4YjJiNGUyNjBkMzQwN2YwNGVlYzllOTVmZTI5ODI4OTZjYTNjYTgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ONsHYM3BUW6NuC5wHxpNaQI8qHT7IKQDobZNGs4jtDQ)
> 
> Just make the callback for the empty state mandatory, and _always_ render the button. I.e., we dont need to conditionally render it like this: ![immagine](https://private-user-images.githubusercontent.com/62880920/435406855-f0c675c1-dcfc-438a-a8f7-e79c92ecf957.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUxMzEwNDMsIm5iZiI6MTc0NTEzMDc0MywicGF0aCI6Ii82Mjg4MDkyMC80MzU0MDY4NTUtZjBjNjc1YzEtZGNmYy00MzhhLWE4ZjctZTc5YzkyZWNmOTU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIwVDA2MzIyM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU4M2MxNWQ0OWJjOWE0MDUwMzMwZWU3N2U5NTVlN2QzYmZkOWZiNDNmMzFjMGMwMzAyNThjYzAzZjc0ZWVmYWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.igE0ADsgRusYClJdrgKCtKp6SuAGF02RQeMwG_uugPM) Just show it always
> 
> @ObitoDevv @anilrise11 if you could re-do a quality control of all the code in this PR to prevent this kind of issues (unnecessary complexity, stories that don't align with the intended use of the components, images that show alt texts) that would be very helpful!

Hello @alebg Thanks for pointing out the issues. We are going through issues you have pointed out and entire code again and will send you further updates in a while...

ObitoDevv commented:
Hi @aleg sir,

Here’s a concise overview of the changes made as per our discussion:

EmptyState Refactor

- Replaced CourseEmptyState with a new EmptyState component.
- Added {buttonText?.trim() && buttonText && <Button ... />} to avoid rendering empty/whitespace buttons.
- Made message mandatory in EmptyState; buttonText and onButtonClick are optional.
- Updated CourseCardList stories and tests to use EmptyState with locale support (en, de).
- Removed obsolete CourseEmptyState stories/tests.

UserType Handling

- Avoided userType-based logic in EmptyState since CourseCard already handles it, letting the platform manage empty state props.

Other Fixes

- Fixed coach banner alt-text and image issues.
- Corrected hover and overflow for long course titles.
- Updated Storybook for accurate rendering/documentation.

Let me know if you need more details!

alebg reviewed (changes requested):
@ObitoDevv thanks a lot, it's looking much better. So now, another thing I've noticed, is that the empty state of the course card list doesn't expand horizontally. Is this a problem of the story? Or is there some style conflicting with how this should look?

![immagine](https://github.com/user-attachments/assets/3f1c755f-c271-4f64-aad7-5945c7a8e5f4)

The story of the isolated empty state is fine though.

ObitoDevv commented:
Hi @aleg sir, no changes needed! The empty state not expanding horizontally in Storybook is due to it lacking a parent to inherit width from. In the actual page, the Empty State Component will take the full width of its parent container. In Storybook, without a parent, it defaults to w-auto. It'll work fine in the page context.

carlapbg reviewed (dismissed):
Looks good now, thanks!

alebg commented:
@ObitoDevv thanks a lot! Note the commit I sent, in which I renamed files to be consistent, applied some formatting, and removed unused imports

alebg commented:
Will create a merge PR -- this is a very old PR and rebasing is too messy.
